### ISSUE TITLE
Filtering by lower vibrational and/or rotational states and/or delta nu and/or delta J

### ISSUE THREAD

**Issue by HajimeKawahara** [HajimeKawahara/exojax#330]
For a deeper analysis of exoplanet atmosphere, we would implement filtering of lower vibrational and/or rotational states and/or delta nu and/or delta J when calling MdbExomol instance.

I am not sure if this can be done in `radis.api`. @erwanp 
If not we directly edit `radis.api`.

**Comment by HajimeKawahara** [comment]
Just note:
ExoMol: Current implementation of mdb DataFrame , i.e. in ExoMol .trans file, contains
-  i_upper	i_lower	A	nu_lines	gup	jlower	jupper	elower	Si

We need to connect i_upper and i_lower (state counting number) to quantum states, defined in .states file. 
Because ExoJAX already moved the ExoMol I/O to radis.api, this should be done in radis.api, as far as I understand. 
I think this is the same procedure that extracts J numbers from the .states file, using [map_add](https://github.com/radis/radis/blob/27a80617bb2675ef8b7c88e549c430ef947418c2/radis/api/exomolapi.py#L458), e.g.  [here](https://github.com/radis/radis/blob/27a80617bb2675ef8b7c88e549c430ef947418c2/radis/api/exomolapi.py#L485).

![exomol_states](https://user-images.githubusercontent.com/15956904/213895055-dce68d19-a2b4-46cb-923e-f97a09169c29.png)

Reference: Table 5 in https://arxiv.org/abs/1801.09803

``` python
# %%
from exojax.spec import api
from exojax.utils.grids import wavenumber_grid
nus,wav,r = wavenumber_grid(22900.0,24000.0,1000,unit="AA")
emf='CO/12C-16O/Li2015'
mdb = api.MdbExomol(emf,nus,inherit_dataframe=True)

# %%
mdb.df
# Current column
# i_upper	i_lower	A	nu_lines	gup	jlower	jupper	elower	Si
```

**Comment by HajimeKawahara** [comment]
- example (Li2015)
In .def
```
v                                                                               # Quantum label 1
I4 %4d                                                                          # Format quantum label 1
State vibrational quantum number                                                # Description quantum label 1
kp                                                                              # Quantum label 2
A4 %4s                                                                          # Format quantum label 2
kronig rotationless parity                                                      # Description quantum label 2

```
So, 

```
shirochan:~/exojax/documents/analysis/CO/12C-16O/Li2015/tmp(select_nuJ)>more 12C-16O__Li2015.states 
           1     0.000000      1       0       0   e
           2  2143.271100      1       0       1   e
```
means (i, E, g, J, v, kp)

**Comment by HajimeKawahara** [comment]
`skip_optional_data = False` loads other entries, but looks something wrong. vl and kp_l shifted. I will fix this bug. 
```
default broadening parameters are used for  71  J lower states in  152  states
#        i_upper    i_lower    A          nu_lines      gup    jlower    jupper    elower      v_l    kp_l    Sij0
0        84         42         1.155e-06  2.405586      3      0         1         66960.7124  e      nan     3.811968898414225e-164
1        83         41         1.161e-06  2.441775      3      0         1         65819.903   e      nan     9.663028103692631e-162
2        82         40         1.162e-06  2.477774      3      0         1         64654.9206  e      nan     2.7438392479197905e-159
```

**Comment by HajimeKawahara** [comment]
A nice part already implemented in radis (which I did not know), is to load the other entries than the mandatory field (i, E, g, J), :
```
{'n_Texp': 0.5, 'alpha_ref': 0.07, 'molmass': 28.0101, 'numinf': None, 'numtag': '', 'quantum_labels': ['v', 'kp'], 'Landé': False, 'lifetime': False}
```
I can start from this!

**Comment by HajimeKawahara** [comment]
Check done for
- [ ] MgH, (Lande, lifetime, uncertainty of the line center=unc)
- [ ] NaH, 
- [ ] CO

**Comment by erwanp** [comment]
It should be able a user-defined key and not hardcoded ; because I expect the States to change with each molecules and in the future, we don't want to release a new version each time

**Comment by HajimeKawahara** [comment]
Yeah, sure. It's not hardcoded. See [here](https://github.com/radis/radis/blob/97b811a38a7c21f00f9a4e49e6f5faa70ce1f908/radis/api/exomolapi.py#L347).  I just wanted to check if the code works for major species. 

Actually, loading states based on the def file is a bit tricky. Some note: It looks
- if unc (uncertainty of line center),lifetime, Lande exists, additional row(s) should be placed before `quantum_labels`
- some additional rows (maybe not described in def) can exist after  `quantum_labels` (I found e.f. Marvelized/Calculated, Calculated energy)

I am checking if this rule is general for most species. @erwanp

**Comment by HajimeKawahara** [comment]
Actually, the above rule looks general for most species.

**Comment by HajimeKawahara** [comment]
I found some exception:
HCl (HCl/1H-35Cl/HITRAN-HCl/) has electronic states (eS) in .def, but not in states. it should be handled as an exception. 

in def
```
2                                                                               # No. of quanta defined
v                                                                               # Quantum label 1
I4 %4d                                                                          # Format quantum label 1
State vibrational quantum number                                                # Description quantum label 1
eS                                                                              # Quantum label 2
A4 %4s                                                                          # Format quantum label 2
electronic state                                                                # Description quantum label 2
```

```
shirochan:~/exojax/documents/analysis/tmp(select_nuJ)>more 1H-35Cl__HITRAN-HCl.states 
           1     0.000000      8       0       0
```
I added the exception in `read_def` and it worked.

**Comment by HajimeKawahara** [comment]
done.

**Comment by HajimeKawahara** [comment]
addressed by #336
